infra/vsts_agent_ubuntu_20_04_startup.sh

```diff
@@ -95,6 +95,44 @@ useradd \
   vsts
 #add docker group to user
 usermod -aG docker vsts
+# let vsts user mount/unmount cache folders
+echo "/tmp/bazel_cache /home/vsts/.cache/bazel auto rw,user,exec" >> /etc/fstab
+echo "/tmp/disk_cache /home/vsts/.bazel-cache auto rw,user,exec" >> /etc/fstab
+
+CACHE_SCRIPT=/home/vsts/reset_caches.sh
+
+cat <<'RESET_CACHES' > $CACHE_SCRIPT
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+reset_cache() {
+    local file mount_point
+    file=$1
+    mount_point=$2
+
+    echo "Cleaning up '$mount_point'..."
+    if [ -d "$mount_point" ]; then
+        for pid in $(pgrep -a -f bazel | awk '{print $1}'); do
```

Comment:
What happens if there are no PIDs?

Reply:
`for` doesn't run if the list to run over is empty, like in most languages.